src/mcp_server_motherduck/server.py

```diff
@@ -88,25 +112,37 @@ def _resolve_db_path_type(
             )
         return db_path, "duckdb"
 
+    def _execute(self, query: str) -> str:
+        if self.conn is None:
+            # open short lived readonly connection, run query, close connection, return result
+            conn = duckdb.connect(
+                self.db_path,
+                config={"custom_user_agent": f"mcp-server-motherduck/{SERVER_VERSION}"},
+                read_only=True,
```

Comment:
```suggestion
                read_only=self._read_only,
```

Just to make it explicit